instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.net;
```

Comment:
I guess this should be in javaagent-api since it's only for auto instrumentation?

Reply:
That's the problem: if I put it inside javaagent-api then I'd have to make library instrumentations (armeria, okhttp, ...) depend on javaagent-api to use that (because `NetAttributeExtractor` implementations are package-private and not visible from the javaagent code).

Reply:
Oops package names are so long so got confused. Hmm - yeah we definitely shouldn't use this code from library instrumentation, libraries are just supposed to be called with something like `.addAttributeExtractor(FixedAttributeExtractor.of(PEER_SERVICE, "my-service"))`, there's no point of the map approach outside of agent and it shouldn't be supported.

Should we just access the package private `ArmeriaNetAttributesExtractor` from the agent reflectively?

Reply:
> Should we just access the package private `ArmeriaNetAttributesExtractor` from the agent reflectively?

Hmm, that would work too -- I'll implement that option in this PR and we'll see how it looks.